pandas/core/internals/construction.py

```diff
@@ -339,7 +354,7 @@ def _homogenize(data, index, dtype=None):
     return homogenized
 
 
-def extract_index(data):
+def extract_index(data) -> Index:
```

Comment:
in the absence of type annotations for data, can you add a docstring (one-liner ok as a starter) to help future readers.